calico/operations/ebpf/troubleshoot-ebpf.mdx

```diff
@@ -69,114 +67,158 @@ The tool is embedded in the $[nodecontainer] container image. To run the tool:
     conntrack    Manipulates connection tracking
     counters     Show and reset counters
     help         Help about any command
+    ifstate      Manipulates ifstate
     ipsets       Manipulates ipsets
     nat          Manipulates network address translation (nat)
+    policy       Dump policy attached to interface
+    profiling    Show and reset profiling data
     routes       Manipulates routes
     version      Prints the version and exits
 
-  Flags:
-    --config string   config file (default is $HOME/.calico-bpf.yaml)
-    -h, --help            help for calico-bpf
-        --log-level string   Set log level (default "warn")
-    -t, --toggle          Help message for toggle
-  ```
-
   (Since the tool is embedded in the main `calico-node` binary the `--help` option is not available, but running
   `calico-node -bpf help` does work.)
 
-  To dump the BPF conntrack table:
+  For example, to dump the BPF conntrack table use:
 
   ```
   kubectl exec -n calico-system calico-node-abcdef -- calico-node -bpf conntrack dump
   ...
   ```
 
-  Also, it is possible to fetch various counters, like packets dropped by a policy or different errors, from BPF data plane using the same tool.
-  For example, to dump the BPF counters of `eth0` interface:
+## Debug access to services
+
+  Inspect the BPF NAT table to verify that the service is correctly programmed. To dump the BPF NAT table:
 
   ```
-  kubectl exec -n calico-system calico-node-abcdef -- calico-node -bpf counters dump --iface=eth0
-  +----------+--------------------------------+---------+--------+-----+
-  | CATEGORY |              TYPE              | INGRESS | EGRESS | XDP |
-  +----------+--------------------------------+---------+--------+-----+
-  | Accepted | by another program             |       0 |      0 |   0 |
-  |          | by failsafe                    |       0 |      2 |  23 |
-  |          | by policy                      |       1 |      0 |   0 |
-  | Dropped  | by policy                      |       0 |      0 |   0 |
-  |          | failed decapsulation           |       0 |      0 |   0 |
-  |          | failed encapsulation           |       0 |      0 |   0 |
-  |          | incorrect checksum             |       0 |      0 |   0 |
-  |          | malformed IP packets           |       0 |      0 |   0 |
-  |          | packets with unknown route     |       0 |      0 |   0 |
-  |          | packets with unknown source    |       0 |      0 |   0 |
-  |          | packets with unsupported IP    |       0 |      0 |   0 |
-  |          | options                        |         |        |     |
-  |          | too short packets              |       0 |      0 |   0 |
-  | Total    | packets                        |      27 |    124 |  41 |
-  +----------+--------------------------------+---------+--------+-----+
-  dumped eth0 counters.
+  kubectl exec -n calico-system calico-node-abcdef -- calico-node -bpf nat dump
```

Comment:
Suggestion, let's use the DS make it more copy pastable.
```bash
kubectl exec -n calico-system ds/calico-node -- calico-node -bpf nat dump
```

Reply:
TIL that this was possible.  Since these are troubleshooting steps though, I think the user may want to be specific about which node they're looking at (e.g. where their application is).